src/coreclr/tools/Common/TypeSystem/Common/ExplicitLayoutValidator.cs

```diff
@@ -15,22 +16,64 @@ private enum FieldLayoutTag : byte
             ORef,
         }
 
+        private struct FieldLayoutInterval : IComparable<FieldLayoutInterval>
+        {
+            public FieldLayoutInterval(int start, int size, FieldLayoutTag tag)
+            {
+                Start = start;
+                Size = size;
+                Tag = tag;
+            }
+
+            public int Start;
+            public int Size;
+
+            public int EndSentinel
+            {
+                get
+                {
+                    return Start + Size;
+                }
+                set
+                {
+                    Size = value - Start;
+                    Debug.Assert(Size >= 0);
+                }
+            }
+
+            public FieldLayoutTag Tag;
+
+            public int CompareTo(FieldLayoutInterval other)
+            {
+                if (other.Start == Start)
```

Comment:
Non-blocking nit - looks like it might suffice to <code>return Start.CompareTo(other.Start)</code>.

Reply:
Yeah, that's a bit more obvious.